packages/flutter_tools/lib/src/project.dart

```diff
@@ -174,22 +177,47 @@ class IosProject {
   /// The 'Manifest.lock'.
   File get podManifestLock => directory.childDirectory('Pods').childFile('Manifest.lock');
 
+  /// The 'Info.plist' file of the host app.
+  File get hostInfoPlist => directory.childDirectory(_hostAppBundleName).childFile('Info.plist');
+
   /// '.xcodeproj' folder of the host app.
   Directory get xcodeProject => directory.childDirectory('$_hostAppBundleName.xcodeproj');
 
   /// The '.pbxproj' file of the host app.
   File get xcodeProjectInfoFile => xcodeProject.childFile('project.pbxproj');
 
-  /// The product bundle identifier of the host app.
+  /// The product bundle identifier of the host app, or null if not set or if
+  /// iOS tooling needed to read it is not installed.
   String get productBundleIdentifier {
-    return _firstMatchInFile(xcodeProjectInfoFile, _productBundleIdPattern)?.group(1);
+    final String fromPbxproj = _firstMatchInFile(xcodeProjectInfoFile, _productBundleIdPattern)?.group(1);
+    final String fromPlist = iosWorkflow.getPlistValueFromFile(
```

Comment:
Perhaps check this first to avoid looking in pbxproj if not needed

Reply:
Done.